java/com/google/re2j/Parser.java

```diff
@@ -980,6 +983,7 @@ private Regexp parseInternal() throws PatternSyntaxException {
     if (n != 1) {
               throw new PatternSyntaxException(ERR_MISSING_PAREN, wholeRegexp);
     }
+    stack.get(0).namedGroups = namedGroups;
```

Comment:
This looks wrong.

This is what they do in c++: 
https://github.com/google/re2/blob/master/re2/regexp.cc#L604-L650

This is what I ported in Scala
https://github.com/scala-native/re2s/commit/9a40319b21629b27b4fb2b9685cdc6428da26003#diff-3a6eea39d89568034dd070ba707f60d9R236

Reply:
Not sure it's wrong, just different from the C++ or Scala impls of RE2. The approach in my change appears to be similar to that in the Java SDKs, see http://www.docjar.com/html/api/java/util/regex/Pattern.java.html around line 2800.

Reply:
@sjamesr you cannot look at the code in OpenJDK. It's Licensed under GPL. This project is Licensed under this custom License. https://github.com/google/re2j/blob/master/LICENSE